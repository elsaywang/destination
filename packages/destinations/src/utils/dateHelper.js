import moment from 'moment';

export const datePattern = 'MMM DD,YYYY hh:mm a';

export const formatDate = ms => moment(ms).format(datePattern);

export const isExpired = ms => moment(ms).isBefore(moment());

export const expireIn = ms => isExpired(ms) ? 'Expired' : moment(ms).diff(moment(), "days")+` days`;
