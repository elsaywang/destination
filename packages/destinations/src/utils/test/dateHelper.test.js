import moment from 'moment';
import { datePattern, formatDate, isExpired, expireIn } from '../dateHelper';
import mockdate from 'mockdate';

describe('dateHelper until test', () => {
    beforeEach(() => {
        mockdate.set(1525176000000); // Mon May 01 2018 12:00:00 GMT+0000 (GMT)
    });

    afterEach(() => {
        mockdate.reset();
    });

    it('matches snapshot', () => {
        expect(datePattern).toMatchSnapshot();
    });
    it(`formatDate func test`, () => {
        expect(formatDate(1548345128000)).toEqual('Jan 24, 2019 03:52 PM');
    });
    it(`isExpired func test`, () => {
        expect(isExpired(1548345128000)).toBeFalsy();
        expect(isExpired(1525175000000)).toBeTruthy();
    });

    it(`expireIn func test`, () => {
        expect(expireIn(1548345128000)).toEqual('268 days');
        expect(expireIn(1525175000000)).toEqual('Expired');
    });
});
