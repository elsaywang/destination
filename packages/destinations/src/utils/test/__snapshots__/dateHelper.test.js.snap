// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateHelper until test matches snapshot 1`] = `"MMM DD, YYYY hh:mm A"`;
